src/event-helper.js

```diff
@@ -154,19 +156,29 @@ export function loadPromise(eleOrWindow, opt_timeout) {
       unlistenError = listenOnce(eleOrWindow, 'error', reject);
     }
   });
-  loadingPromise = loadingPromise.then(() => eleOrWindow, failedToLoad);
-  return racePromise_(loadingPromise, unlistenLoad, unlistenError,
-      opt_timeout);
+
+  return loadingPromise.then(() => {
+    if (unlistenError) {
+      unlistenError();
+    }
+    return eleOrWindow;
+  }, () => {
+    if (unlistenLoad) {
+      unlistenLoad();
+    }
+    failedToLoad(eleOrWindow);
```

Comment:
`failedToLoad` throws an exception, right? Not a huge fun of those methods. Could you please add a comment here and say that this method will rethrow the original error?